Issue #3222835: Basic page's small_teaser display mode references layout builder settings.

mparker17 opened the pull request:
## Problem

After commit f11bf26e47c93af7bee59592e753f8380b9876a7 (whose commit message references [#3193302](https://www.drupal.org/project/social/issues/3193302)); `modules/social_features/social_page/config/install/core.entity_view_display.node.page.small_teaser.yml` contains layout_builder config in the third_party_settings section, which causes the following issue when you try to do a fresh site install without layout_builder enabled:

```
Drupal\Core\Config\ConfigImporterException: There were errors validating the config synchronization.
Configuration <em class="placeholder">core.entity_view_display.node.page.small_teaser</em> depends on the <em class="placeholder">Layout Builder</em> module that will not be installed after import. in
/app/web/core/lib/Drupal/Core/Config/ConfigImporter.php:755
Stack trace:
#0 /app/web/core/lib/Drupal/Core/Config/ConfigImporter.php(539): Drupal\Core\Config\ConfigImporter->validate()
#1 /app/web/core/lib/Drupal/Core/Config/ConfigImporter.php(485): Drupal\Core\Config\ConfigImporter->initialize()
#2 /app/web/modules/contrib/config_enforce/src/ConfigImporter.php(45): Drupal\Core\Config\ConfigImporter->import()
#3 /app/web/modules/contrib/config_enforce/src/ConfigEnforcer.php(100): Drupal\config_enforce\ConfigImporter->importConfig(Array)
#4 /app/web/modules/contrib/config_enforce/src/ConfigEnforcer.php(62): Drupal\config_enforce\ConfigEnforcer->readRegistriesFromDisk()
#5 /app/web/modules/contrib/config_enforce/config_enforce.module(38): Drupal\config_enforce\ConfigEnforcer->enforceConfigs()
#6 [internal function]: config_enforce_rebuild()
#7 /app/web/core/lib/Drupal/Core/Extension/ModuleHandler.php(403): call_user_func_array('config_enforce_...', Array)
#8 /app/web/core/includes/common.inc(1082): Drupal\Core\Extension\ModuleHandler->invokeAll('rebuild')
#9 /app/web/core/includes/utility.inc(55): drupal_flush_all_caches()
#10 /app/vendor/drush/drush/commands/core/cache.drush.inc(302): drupal_rebuild(Object(Composer\Autoload\ClassLoader), Object(Symfony\Component\HttpFoundation\Request))
#11 /app/vendor/drush/drush/includes/command.inc(422): drush_cache_rebuild()
#12 /app/vendor/drush/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#13 /app/vendor/drush/drush/includes/command.inc(199): drush_command()
#14 /app/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#15 /app/vendor/drush/drush/includes/preflight.inc(67): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#16 /app/vendor/drush/drush/drush.php(12): drush_main()
#17 {main}
```

## Solution

Empty the third_party_settings part of the config.

## Issue tracker

https://www.drupal.org/project/social/issues/3222835

navneet0693 commented:
Hey @mparker17 ,

Thanks for your contribution, what are the steps which are you taking for installing the site?

mparker17 commented:
@navneet0693

To some extent, regardless of how I install it, [social_page.info](https://github.com/goalgorilla/open_social/blob/10.3.x/modules/social_features/social_page/social_page.info.yml) doesn't list `layout_builder` as a dependency; and yet [`core.entity_view_display.node.page.small_teaser.yml`](https://github.com/goalgorilla/open_social/blob/10.3.x/modules/social_features/social_page/config/install/core.entity_view_display.node.page.small_teaser.yml) does... and the error indicates that core's configuration management subsystem doesn't just ignore third_party_settings that belong to modules which aren't enabled. Note that the layout_builder settings causing the problem (which I've deleted in this patch) also don't do anything. 

All of this indicates to me that --- regardless of _how_ I'm installing the site --- the layout_builder settings shouldn't be there.

But to directly answer your question, I've been installing OpenSocial with a custom install profile that inherits from the OpenSocial install profile (made possible via the core patch in [#1356276-603](https://www.drupal.org/project/drupal/issues/1356276#comment-13405100)). Our custom install profile installs `social_page` (but not `layout_builder`). This error shows up when installing this way. (as a bit of context, the site itself was initially installed as OpenSocial 6, and its content was migrated to OS8, then upgraded to OS9 then upgraded to OS10 --- we're only doing the fresh install that triggers this error in our CI process so we don't have to send a database to our CI server to run Behat tests for the functionality we added on top of OpenSocial).

Admittedly, when I tried installing a fresh version of OpenSocial just now (i.e.: with `composer create-project goalgorilla/social_template:dev-master example --no-interaction && cd example && drush -y si social`), I don't see the error. I do see that `social_page` gets enabled, and `layout_builder` does not like our custom profile.

However, given that I can clearly see the unstated config dependency on layout_builder, I suspect some other mechanism is causing the error not to show up when I install a fresh version of OpenSocial. My gut says it could be that Drupal just so happens to install `social_page` before the `small_teaser` view mode; or that Drupal isn't strictly checking for config issues at the point in the install process when this config gets imported.

navneet0693 commented:
Thanks for responding @mparker17 !! This helped me understand this issue better. I have created a follow-up PR using the changes from this PR: https://github.com/goalgorilla/open_social/pull/2467. I will close this one. Proper credits will definitely be given.
